test/unit/range/iterator_test_template.hpp

```diff
@@ -0,0 +1,454 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+#pragma once
+
+#include <gtest/gtest.h>
+
+#include <seqan3/core/type_traits/range.hpp>
+#include <seqan3/range/concept.hpp>
+
+using namespace seqan3;
+
+template <typename T>
+struct iterator_fixture : public ::testing::Test
+{
+    /* Please provide the following members:
+
+    using iterator_tag = ...                                   // Defines the iterator functionality you want to test.
+                                                               // One of:
+                                                               // std::forward_iterator_tag
+                                                               // std::bidirectional_iterator_tag
+                                                               // std::random_access_iterator_tag
+
+    static constexpr bool const_iterable = true/false;         // Also test const_iterability.
+
+    range_t range_to_compare;                                  // Used to compare the iterator range with (needs to be
+                                                               // a Foward/RandomAccessRange depending on the iterator).
+
+    T begin_iterator;                                          // Begin of the iterator (range) to test.
+    T end_iterator;                                            // End of the iterator (range) to test.
+
+    // Only need to be specified if const_iterable == true.
+    const_T begin_const_iterator;                               // Begin of the const iterator (range) to test.
+    const_T end_const_iterator;                                 // end of the const iterator (range) to test.
+
+    */
+};
+
+TYPED_TEST_CASE_P(iterator_fixture);
+
+TYPED_TEST_P(iterator_fixture, concept_check)
+{
+    if constexpr (std::Same<typename TestFixture::iterator_tag, std::forward_iterator_tag>)
+    {
+        EXPECT_TRUE(std::ForwardIterator<TypeParam>);
+
+        if constexpr (TestFixture::const_iterable)
+        {
```

Comment:
I thought we agreed on "if the outer scoper has braces the inner one gets them too".
but I dont care

Reply:
hm the compiler does not like this either:
```
/srv/public/smehringer/seqan3/test/unit/range/detail/../iterator_test_template.hpp:78:22: error: suggest explicit braces to avoid ambiguous ‘else’ [-Werror=dangling-else]
         if constexpr (TestFixture::const_iterable)
```

Reply:
This is weird 😕 but keep it then as is